doc(storage-network): support RWX volumes (#911)
longhorn/website: content/docs/1.7.0/references/settings.md

> **Note:** This setting doesn't apply to below cases.
> - The workload pods don't have a controller; Longhorn never deletes them.

> > - The workload pods with *cluster network* RWX volumes. This is because the Longhorn share manager, which provides the RWX NFS service, has its own resilience mechanism to ensure availability until the volume gets reattached without relying on the pod lifecycle to trigger volume reattachment. For details, see [here](../../nodes-and-volumes/volumes/rwx-volumes). However, this setting applies to the workload pods with *storage network* RWX volumes. For more information, see [here](../../advanced-resources/deploy/storage-network#limitation).

Reviewer: ```suggestion
> - Workload pods with *cluster network* RWX volumes. The setting does not apply to such pods because the Longhorn Share Manager, which provides the RWX NFS service, has its own resilience mechanism. This mechanism ensures availability until the volume is reattached without relying on the pod lifecycle to trigger volume reattachment.  The setting does apply, however, to workload pods with *storage network* RWX volumes. For more information, see [ReadWriteMany (RWX) Volume](../../nodes-and-volumes/volumes/rwx-volumes) and [Storage Network](../../advanced-resources/deploy/storage-network#limitation).
```

The storage network uses Multus NetworkAttachmentDefinition to segregate the in-cluster data traffic from the default Kubernetes cluster network.

By default, the this setting applies only to RWO (Read-Write-Once) volumes. For RWX (Read-Write-Many) volumes, see [Storage Network for RWX Volume Enabled](#storage-network-for-rwx-volume-enabled) setting.

> **Warning:** This setting should change after all Longhorn volumes are detached because some pods that run Longhorn system components are recreated to apply the setting. When all volumes are detached, Longhorn attempts to restart all Instance Manager and Backing Image Manager pods immediately. When volumes are in use, Longhorn components are not restarted, and you need to reconfigure the settings after detaching the remaining volumes; otherwise, you can wait for the setting change to be reconciled in an hour.

See [Storage Network](../../advanced-resources/deploy/storage-network) for details.

#### Storage Network For RWX Volume Enabled

> Default: `false`

This setting allows Longhorn to use the storage network for RWX volumes.

> > **Warning:** This setting should change after all Longhorn RWX volumes are detached because some pods that run Longhorn system components are recreated to apply the setting. When all RWX volumes are detached, Longhorn attempts to restart all CSI plugin pods immediately. When volumes are in use, Longhorn components are not restarted, and you need to reconfigure the settings after detaching the remaining volumes; otherwise, you can wait for the setting change to be reconciled in an hour.
> 
> > **Warning:** The RWX volumes will be mounted with the storage network within the CSI plugin pod container network namespace. As result, restarting the CSI plugin pod may lead to unresponsive RWX volume mounts. If this occurs, you will need to restart the workload pod to re-establish the mount connection. Alternatively, you can enable the [Automatically Delete Workload Pod when The Volume Is Detached Unexpectedly](#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly) setting.
> 
> See [Storage Network](../../advanced-resources/deploy/storage-network) for details.

Reviewer: ```suggestion
> **Warning:** 
> This setting should change after all Longhorn RWX volumes are detached because some pods that run Longhorn components are recreated to apply the setting. When all RWX volumes are detached, Longhorn attempts to restart all CSI plugin pods immediately. When volumes are in use, pods that run Longhorn components are not restarted, so the settings must be reconfigured after the remaining volumes are detached. If you are unable to manually reconfigure the settings, you can opt to wait because settings are synchronized hourly.
> 
> The RWX volumes are mounted with the storage network within the CSI plugin pod container network namespace. As a result, restarting the CSI plugin pod may lead to unresponsive RWX volume mounts. When this occurs, you must restart the workload pod to re-establish the mount connection. Alternatively, you can enable the [Automatically Delete Workload Pod when The Volume Is Detached Unexpectedly](#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly) setting.
> 
> For more information, see [Storage Network](../../advanced-resources/deploy/storage-network).
```

Author: > For more information, see [Storage Network](../../advanced-resources/deploy/storage-network).

I've moved this outside of the warning block, as it doesn't seem like it should be part of the warning.
